pkg/config/parameters.go

```diff
@@ -429,6 +429,9 @@ type ClusterParameters struct {
 	//
 	// +optional
 	PerConnectionBufferLimitBytes *uint32 `yaml:"per-connection-buffer-limit-bytes,omitempty"`
+
+	// UpstreamTLS contains the TLS policy parameters for upstream connections
+	UpstreamTLS TLSParameters `yaml:"upstream-tls,omitempty"`
```

Comment:
Should we avoid reusing the `TLSParameters` struct here since it contains some parameters that can be meaningful only for downstream or upstream TSL? 

For example field `TLSParameters.FallbackCertificate` is used as server certificate, which appears as configuration file field  `tls.fallback-certificate`. When reusing the struct for upstream, it could be also defined as `cluster.upstream-tls.fallback-certificate` according purely to syntax, but if defined there, it would be ignored. 

Similarly, config file field `tls.envoy-client-certificate` is used as optional client cert in upstream TLS. Reusing the struct would mean field `cluster.upstream-tls.envoy-client-certificate` is syntactically correct, but actually it will not get used.

Reply:
hmmm I see. I went with TLSParameters originally because it had what I needed and I didn't want to create a similar-but-not-quite-the-same struct.

What would you say to creating a new struct that just contains `{MinimumProtocolVersion,MaximumProtocolVersion,CipherSuites}`, and then making that struct a part of TLSParameters?

Reply:
:+1: that sounds good to me

Reply:
Okay, I ended up creating a new smaller struct called ProtocolParameters. I can always change it if you think a different name is better